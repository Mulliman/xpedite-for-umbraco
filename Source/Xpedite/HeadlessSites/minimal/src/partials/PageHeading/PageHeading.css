.partial-page-heading {
    border: 2px solid #C66476;
}

.partial-page-heading-name {
    background: #C66476;
    color: white;
}