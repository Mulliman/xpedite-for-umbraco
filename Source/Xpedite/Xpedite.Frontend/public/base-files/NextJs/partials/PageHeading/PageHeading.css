.partial-page-heading {
    border: 2px solid #2A9088;
}

.partial-page-heading-name {
    background: #2A9088;
    color: white;
}