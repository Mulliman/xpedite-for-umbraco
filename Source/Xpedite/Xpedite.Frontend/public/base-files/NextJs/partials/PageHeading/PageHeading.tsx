import { TypedUmbracoNode } from "@/umbraco/types";

import './PageHeading.css'

export type PageHeadingProps = { 
    pageTitle?: any,
}

export default function PageHeading ({ pageTitle } : PageHeadingProps) {
    return (
        <div className="partial-page-heading">
            <div className="partial-page-heading-name">Partial: PageHeading</div>
    
            <h1>{ pageTitle }</h1>
        </div>
    )
}