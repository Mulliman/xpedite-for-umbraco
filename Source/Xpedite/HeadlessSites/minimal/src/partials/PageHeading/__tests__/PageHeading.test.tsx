import React from 'react';
import { render } from '@testing-library/react';
import PageHeading, { PageHeadingProps  } from '../PageHeading';

describe('PageHeading', () => {

    const testData = {
        pageTitle: "Hello, this is a test"
    } as PageHeadingProps;

    const emptyTestData = {
        // Fill in props with with the empty state
    } as PageHeadingProps;

    it('matches snapshot', () => {
        const { asFragment } = render(<PageHeading {...testData} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('contains expected element example', () => {
        const { container } = render(<PageHeading {...testData} />);
        expect(container.innerHTML).toContain(`<h1>${testData.pageTitle}</h1>`);
    });

    it('renders valid output when properties are empty', () => {
        const { container } = render(<PageHeading {...emptyTestData} />);
        expect(container.innerHTML).toContain('<h1></h1>');
    });
});