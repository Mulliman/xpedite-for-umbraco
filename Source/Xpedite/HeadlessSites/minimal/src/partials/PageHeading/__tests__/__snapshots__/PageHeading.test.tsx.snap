// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeading matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="partial-page-heading"
  >
    <div
      class="partial-page-heading-name"
    >
      Partial: PageHeading
    </div>
    <h1>
      Hello, this is a test
    </h1>
  </div>
</DocumentFragment>
`;
